src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationService.java

```diff
@@ -13,34 +13,52 @@
 import uk.gov.justice.hmpps.prison.api.model.adjudications.AdjudicationSummary;
 import uk.gov.justice.hmpps.prison.api.model.adjudications.Award;
 import uk.gov.justice.hmpps.prison.api.model.adjudications.Hearing;
+import uk.gov.justice.hmpps.prison.api.model.adjudications.OffenderAdjudicationHearing;
 import uk.gov.justice.hmpps.prison.api.model.adjudications.ProvenAdjudicationSummary;
 import uk.gov.justice.hmpps.prison.api.support.Page;
+import uk.gov.justice.hmpps.prison.api.support.TimeSlot;
 import uk.gov.justice.hmpps.prison.repository.AdjudicationsRepository;
 import uk.gov.justice.hmpps.prison.repository.AgencyRepository;
 import uk.gov.justice.hmpps.prison.repository.LocationRepository;
 import uk.gov.justice.hmpps.prison.security.VerifyBookingAccess;
 import uk.gov.justice.hmpps.prison.security.VerifyOffenderAccess;
 import uk.gov.justice.hmpps.prison.service.support.LocationProcessor;
+import uk.gov.justice.hmpps.prison.util.CalcDateRanges;
 
 import java.time.LocalDate;
 import java.time.temporal.ChronoUnit;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.groupingBy;
 import static java.util.stream.Collectors.toList;
 import static uk.gov.justice.hmpps.prison.repository.support.StatusFilter.ALL;
 
 @Service
 @Transactional(readOnly = true)
-@RequiredArgsConstructor
 public class AdjudicationService {
 
     private final AdjudicationsRepository repository;
     private final AgencyRepository agencyRepository;
     private final LocationRepository locationRepository;
+    @Value("${batch.max.size:1000}")
```

Comment:
There are actually 2 adjudication services.

I went with this one because it is being used by the Offender controller already which is where the new endpoint lives.

Happy to change to use the AdjudicationsService if that is the preferred option.